src/redisearch_rs/rqe_iterators/src/not_iterator.rs

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! Supporting types for [`NotIterator`].
+
+use ffi::t_docId;
+use inverted_index::RSIndexResult;
+
+use crate::{
+    RQEIterator, RQEIteratorError, RQEValidateStatus, SkipToOutcome, maybe_empty::MaybeEmpty,
+};
+
+/// An iterator that negates the results of its child iterator.
+///
+/// Yields all document IDs from 1 to `max_doc_id` (inclusive) that are **not**
+/// present in the child iterator.
+pub struct NotIterator<'index, I> {
+    /// The child iterator whose results are negated.
+    child: MaybeEmpty<I>,
+    /// The maximum document ID to iterate up to (inclusive).
+    max_doc_id: t_docId,
+    /// A reusable result object to avoid allocations on each `read` call.
+    result: RSIndexResult<'index>,
+    // TODO: Timeout
```

Comment:
Looking at the C code, you just need to chain up if the child returned `ITERATOR_TIMEOUT`, right?
This should already be implemented as the `?` operator will return `RQEIteratorError::TimedOut` to the caller.

Or is there something else that I missed?